### PR TITLE
jenkinsfiles: remove kubernetes upstream

### PR DESCRIPTION
**Pull request by aanm**
Now that we are running kubernetes upstream tests as github actions, we can remove this jenkinsfile from the tree.

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]

**Comment by nbusseneau** [comment]
Oh seems like we need to update the lint checks:
```
++ git grep -l quay.io/cilium/cilium-runtime: jenkinsfiles/
fatal: ambiguous argument 'jenkinsfiles/': unknown revision or path not in the working tree.
```

File: jenkinsfiles/kubernetes-upstream.Jenkinsfile [hunk 1]
```diff
@@ -1,132 +0,0 @@
...
-            environment {
-                PROMETHEUS_URL="https://metrics.cilium.io/metrics/job/upstream_job"
-                PROMETHEUS=credentials("metrics")
-            }
-
-            steps {
-                sh '''
-                    cd ${TESTDIR}; vagrant ssh k8s1-${K8S_VERSION} -c "
-                        cd /home/vagrant/go/${PROJ_PATH}/test;
-                        ./kubernetes-netperftest.sh '$PROMETHEUS_URL' '$PROMETHEUS_USR' '$PROMETHEUS_PSW'"
```
Comment (tklauser) [inline jenkinsfiles/kubernetes-upstream.Jenkinsfile:119]: I looks like `test/kubernetes-netperftest.sh` (and in turn `contrib/scripts/netperf_reporter.py`) are now unused, so these scripts could probably be removed as well.

Comment (aanm) [reply jenkinsfiles/kubernetes-upstream.Jenkinsfile:119]: good call!

**Review by tklauser — APPROVED** [review]

**Pull request merged**